org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/navigator/BuildFolderViewerFilter.java

```diff
@@ -31,7 +36,25 @@ public boolean select(Viewer viewer, Object parentElement, Object element) {
     }
 
     private boolean isBuildFolder(IResource resource) {
-        return resource instanceof IFolder && CorePlugin.workspaceOperations().isBuildFolder((IFolder) resource);
+        if (resource instanceof IFolder) {
+            return isBuildFolderInPerstentModel((IFolder) resource);
+        } else {
+            return false;
+        }
     }
 
+    public static boolean isBuildFolderInPerstentModel(IFolder folder) {
+        try {
+            IProject project = folder.getProject();
+            Optional<IPath> modelBuildDir = CorePlugin.modelPersistence().loadModel(project).getBuildDir();
+            if (!modelBuildDir.isPresent()) {
+                return false;
+            }
+            IPath relativePath = RelativePathUtils.getRelativePath(project.getFullPath(), folder.getFullPath());
```

Comment:
Turns out it's not necessary to use `RelativePathUtils` as there's `IFolder.getProjectRelativePath()`.